src/sdk/pycli/nnicli/nni_client.py

```diff
@@ -92,48 +72,384 @@ def _create_process(cmd):
+        self.id = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+    
+    def __repr__(self):
+        return "NNITrialResult(parameter:{} value:{} id:{})".format(self.parameter, self.value, self.id)
+
+class NNITrialMetricData:
+    """
+    NNITrialMetricData stores the metric data of a trial job.
+    A trial job may have both intermediate metric and final metric.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the metric data
+
+    Attributes
+    ----------
+    timestamp:
+        time stamp
+    trialJobId:
+        trial id
+    parameterId:
+        parameter id
+    type:
+        metric type, `PERIODICAL` for intermediate result and `FINAL` for final result
+    sequence:
+        sequence number in this trial
+    data:
+        metric data
+    """
+    def __init__(self, json_obj):
+        self.timestamp = None
+        self.trialJobId = None
+        self.parameterId = None
+        self.type = None
+        self.sequence = None
+        self.data = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+
+    def __repr__(self):
+        return "NNITrialMetricData(timestamp:{} trialJobId:{} parameterId:{} type:{} sequence:{} data:{})" \
+            .format(self.timestamp, self.trialJobId, self.parameterId, self.type, self.sequence, self.data)
+
+class NNITrialHyperParameters:
+    """
+    NNITrialHyperParameters stores the hyper parameters of a trial job.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the hyper parameters
+
+    Attributes
+    ----------
+    parameter_id:
+        parameter id
+    parameter_source:
+        parameter source
+    parameters:
+        hyper parameters
+    parameter_index:
+        parameter index
+    """
+    def __init__(self, json_obj):
+        self.parameter_id = None
+        self.parameter_source = None
+        self.parameters = None
+        self.parameter_index = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+
+    def __repr__(self):
+        return "NNITrialHyperParameters(parameter_id:{} parameter_source:{} parameters:{} parameter_index:{})" \
+            .format(self.parameter_id, self.parameter_source, self.parameters, self.parameter_index)
+
+class NNITrialJob:
+    """
+    NNITrialJob stores the information of a trial job.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the hyper parameters
+
+    Attributes
+    ----------
+    id:
+        trial id
+    status:
+        job status
+    hyperParameters:
+        see `NNITrialHyperParameters`
+    logPath:
+        log path
+    startTime:
+        job start time (timestamp)
+    endTime:
+        job end time (timestamp)
+    finalMetricData:
+        see `NNITrialMetricData`
+    parameter_index:
+        parameter index
+    """
+    def __init__(self, json_obj):
+        self.id = None
```

Comment:
I think these all should be properties.

Reply:
I think these `data` should allow users to modify.